compiler/rustc_trait_selection/src/solve/normalize.rs

```diff
@@ -205,10 +205,9 @@ impl<'tcx> FallibleTypeFolder<TyCtxt<'tcx>> for NormalizationFolder<'_, 'tcx> {
     }
 
     fn try_fold_const(&mut self, ct: ty::Const<'tcx>) -> Result<ty::Const<'tcx>, Self::Error> {
-        let reveal = self.at.param_env.reveal();
         let infcx = self.at.infcx;
         debug_assert_eq!(ct, infcx.shallow_resolve(ct));
-        if !needs_normalization(&ct, reveal) {
+        if !ct.has_projections() {
```

Comment:
new solver normalizes opaques even with `Reveal::UserFacing`, small bug fix.

Reply:
With the new solver opaque types are just another kind of alias type without any special behavior, except for their implementation for `NormalizesTo` goals and the fact that the store already defined opaques in a side-table.

The reasons for that is pretty much: I have no idea how to have a 1) sound 2) stable[^1] and 3) performant implementation which does not do so.

Looking only through the issues I opened an issue for, we get
- https://github.com/rust-lang/trait-system-refactor-initiative/issues/74
- https://github.com/rust-lang/trait-system-refactor-initiative/issues/47
- https://github.com/rust-lang/trait-system-refactor-initiative/issues/69

[^1]: i.e. small future changes to the type systems won't break user code